src/core/data/ions/ion_population/ion_population.hpp

```diff
@@ -24,6 +24,7 @@ namespace core
     class IonPopulation
     {
     public:
+        using grid_type                        = typename VecField::grid_type;
```

Comment:
changes in `src/amr/messengers/hybrid_hybrid_messenger_strategy.hpp` need access to the `grid_type`

it's possible there's an alternative approach

possibly redundant here as the vecfield_t is exposed, so we can get it from there

Reply:
oh I think this is a resource manager required thing now

https://github.com/PHAREHUB/PHARE/pull/790/files#diff-48ce40d5edbf328da2a149410e4ca16306bc2e71103411cea41795681c067ebdR17

Reply:
ooooh, this could actually be bad, but I'm not 100% sure
there is no vector stack varaible "in the stack" anymore, so we shouldn't need any GPU vector symbols/lib (which there isn't (yet at least))
but it's still strange to have the type in vecfield, and reference-able
it might be better for the resource manager itself to take the grid_type, as teh "allocator" so to speak